nft-client/src/main.rs

```
@@ -334,14 +370,55 @@ async fn main() -> anyhow::Result<()> {
                 .context("Failed to connect to Node GRPC")?;
 
             let consensus_status = grpc_client.get_consensus_status().await?;
-            let instance_info = grpc_client
-                .get_instance_info(contract.0, &consensus_status.best_block)
+            let contract_context =
+                smart_contracts::ContractContext::new(contract.0, smart_contracts::ReceiveName {
+                    name: String::from("CIS2-NFT.view"),
+                });
+            let invoke_result = grpc_client
+                .invoke_contract(&consensus_status.best_block, &contract_context)
                 .await?;
-            let mut cursor = concordium_contracts_common::Cursor::new(instance_info.model);
-            let state = NFTContractState::deserial(&mut cursor)
+            let model = match invoke_result {
+                smart_contracts::InvokeContractResult::Success {
+                    return_value: Some(bytes),
+                    ..
+                } => bytes.value,
+                _ => bail!("Failed to invoke the contract"),
```

Can you please not ignore the error, and report it back to the user.

---

Since `RejectReason` does not implement Display or Error, I think the most meaningful I can display is the debug representation.

---

Which is fine.